controllers/mcg_bucket.go

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"strings"
+
+	"github.com/kube-object-storage/lib-bucket-provisioner/pkg/apis/objectbucket.io/v1alpha1"
+	noobaav1alpha1 "github.com/noobaa/noobaa-operator/v5/pkg/apis/noobaa/v1alpha1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	McgmsObcNamespace   = "mcgms-obc-namespace"
+	McgmsCacheEnabled   = "mcgms-cache-enabled"
+	DefaultBackingStore = "noobaa-default-backing-store"
+)
+
+func (r *ManagedMCGReconciler) watchBucketClass(object client.Object) {
+	bucketName := object.GetName()
+	annotations := object.GetAnnotations()
+	if _, ok := annotations[McgmsObcNamespace]; ok {
+		r.objectBucketClaim = &noobaav1alpha1.ObjectBucketClaim{}
+		r.bucketClass = &noobaav1alpha1.BucketClass{}
+		r.ctx = context.Background()
+		if r.isOBCExists(object) {
+			r.Log.Info("OBC already exists", "name", bucketName)
+
+			return
+		}
+
+		obc := r.setOBCDesiredState(bucketName, object)
+		_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.objectBucketClaim, func() error {
+			r.Log.Info("Creating OBC ", "name", bucketName)
+			r.objectBucketClaim.Spec = obc.Spec
+
+			return nil
+		})
+		if err != nil {
+			r.Log.Error(err, "Error while creating OBC, OBC not created")
+
+			return
+		}
+		annotations := object.GetAnnotations()
+		if isCacheEnabled, ok := annotations[McgmsCacheEnabled]; ok && isCacheEnabled == "true" {
```

Comment:
Who will add this annotation? The UI (as we talked earlier)?

Reply:
@rexagod yes UI team

cc : @SanjalKatiyar